app/scripts/directives/mobileClientDownloadUrl.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileClientDownloadUrl', {
+    controller: [
+      'APIService',
+      'DataService',
+      MobileClientDownloadUrl
+    ],
+    bindings: {
+      mobileBuild: '<',
+      showPanel: '<'
+    },
+    templateUrl: 'views/mobile-client-download-url.html'
+  });
+
+  function MobileClientDownloadUrl(
+    APIService,
+    DataService
+  ) {
+    var ctrl = this;
+    ctrl.BUILD_DOWNLOAD_ANNOTATION = 'aerogear.org/download-mobile-artifact-url';
+    ctrl.SET_BUILD_DOWNLOAD_ANNOTATION = 'aerogear.org/download-mobile-artifact';
+    var buildsVersion = APIService.getPreferredVersion('builds');
+
+    ctrl.$onChanges = function(changes) {
+      var mobileBuildChanges = changes.mobileBuild && changes.mobileBuild.currentValue;
+      if (mobileBuildChanges && !ctrl.context) {
+        ctrl.context = {namespace:_.get(ctrl, 'mobileBuild.metadata.namespace')};
+      }
+
+      var showPanelChanges = changes.showPanel && changes.showPanel.currentValue;
+      if (showPanelChanges) {
+        var downloadUrl = ctrl.mobileBuild.metadata.annotations[ctrl.BUILD_DOWNLOAD_ANNOTATION];
```

Comment:
I'm not sure this belongs in onChanges, if you load the page with a build that already has a download link generated, it should show the QR code immediately right? This looks like it won't show it until the build is updated somehow.

Reply:
It's worth testing that anyway, and if it doesn't work, putting this logic inside a watch might work better?

Reply:
if the qr code/download link is already generated, it should be available in the annotations which it checks first. So even if you reload the page, the download url should already be available.

Reply:
yeah, correct, this is fine. I have tested it locally.